Makefile

```diff
@@ -184,6 +185,14 @@ LIBC_TOP_HALF_MUSL_SOURCES = \
                  %/cimagf.c %/cimag.c %cimagl.c, \
                  $(wildcard $(LIBC_TOP_HALF_MUSL_SRC_DIR)/complex/*.c)) \
     $(wildcard $(LIBC_TOP_HALF_MUSL_SRC_DIR)/crypt/*.c)
+
+# Add some additional sources when threads are enabled.
+ifeq ($(THREAD_MODEL), posix)
+LIBC_TOP_HALF_MUSL_SOURCES += $(addprefix $(LIBC_TOP_HALF_MUSL_SRC_DIR)/, \
+    thread/pthread_setcancelstate.c \
```

Comment:
I think at the very least you will need to include `thread/__lock.c` here

Reply:
Actually I think you should probably include that majority of the files in the `thread/` directory?  But maybe you want to save that for a followup PR?

Reply:
I tried to include more files here but it was a game of whack-a-mole (i.e., the build failures). There are many other bits that need to be fixed so I'm hoping this PR can be merged with this very limited subset of the pthreads API and future PRs can add more of the API.

Reply:
Sounds like a good plan.

In that case, maybe just remove this block completely for now? (We really need `pthread_setcancelstate` do we?)

Reply:
Yeah, I can just remove it.